server/src/controllers/middleware/getFavorite.js

```diff
@@ -0,0 +1,36 @@
+const { getFavoriteCourse, getUserById } = require('../../database/queries');
+const { favoriteSchema } = require('../../utils');
+
+const getFavorite = async (req, res, next) => {
+  try {
+    const { userId } = req.params;
+    await favoriteSchema.validate({ userId });
+    if (userId > 0) {
+      const checkUser = await getUserById(userId);
+      if (checkUser.rowCount) {
+        const { rows } = await getFavoriteCourse(userId);
+        if (rows[0]) {
+          res.json(rows);
+        } else {
+          res.json({
+            message: 'There is no favorite courses for this user id',
+          });
+        }
+      } else {
+        res.status(404).json({
+          message: 'Invalid id',
+        });
+      }
+    }
+  } catch (err) {
+    if (err.errors) {
+      res.status(404).json({
+        message: err.errors,
+      });
```

Comment:
**It's a good way to handle validation error but, there are some points needed to be refactored** 
here, you want to handle validation error of `userId` so, the status code should be 400 (bad request) and this syntax will be more clear : 
```suggestion
        if (err.name === 'ValidationError') {
      return res
        .status(400)
        .json({ message: 'invalid inputs' });
    }
      

```